snapcraft/plugins/python.py

```diff
@@ -277,17 +287,44 @@ def __init__(self, *, exec_func, runnable, package_dir, env,
         if dependency_links:
             self._extra_pip_args.append('--process-dependency-links')
 
+    def list(self, exec_func=None):
```

Comment:
coming late to the party, but this name is not good. I would expect something closer to the docstring, like get_installed_python_packages. Also, why is this not marked as internal with the _ in front?